code/lib/blocks/src/blocks/Source.tsx

```diff
@@ -88,8 +85,29 @@ const getSnippet = ({
   const code = useSnippet ? snippet : sourceParameters.originalSource || '';
 
   const transformer = transformFromProps ?? sourceParameters.transform;
+  const transformed = useMemo(
+    () => (transformer ? transformer?.(code, storyContext) : code),
+    [code, storyContext, transformer]
+  );
+
+  useEffect(() => {
+    async function getTransformedCode() {
+      setTransformedCode(await transformed);
+    }
+
+    getTransformedCode();
+  }, [transformed]);
+
+  if (sourceParameters.code !== undefined) {
+    return sourceParameters.code;
+  }
+
+  if (typeof transformed === 'object' && typeof transformed.then === 'function') {
```

Comment:
I don't really understand this if block, shouldn't it always return `transformedCode` and never the `transformed` promise which it does in line 92?

Reply:
`transformed` isn't necessarily a promise! The user can still provide just a plain synchronous function which returns a string. In this case, I want to return its return value immediately and don't want to wait a whole lifecycle to trigger the `useEffect`, update the `transformedCode` state and finally return the result of the asynchronous function.